### PR TITLE
Add chat example documentation.

### PR DESCRIPTION
**Pull request by pepoviola**
Hi @davidpdrsn, first let me say thanks for this awesome work!! I was playing with the examples and I think this little changes can help people to easily run the `chat` example.

## Motivation

Allow people to easy run the `chat` example by check the docs in `chat.rs` or with the hint when run 
```
cargo run --example chat
```

## Solution

- Add example of how to run
- Add `required-features`
- Add `chat` to example list.

Thanks!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -80,3 +80,7 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [package.metadata.playground]
 features = ["ws"]
+
+[[example]]
+name = "chat"
+required-features = ["ws"]
```
Comment (davidpdrsn) [inline Cargo.toml:86]: Whoa! I didn't realize you can do this. Thats cool! We should totally add for the other examples as well. Doesn't have to be done in this PR though.

File: examples/chat.rs [hunk 2]
```diff
@@ -1,3 +1,11 @@
+//! Example chat application.
+//!
+//! Run with
+//!
+//! ```
+//!     cargo run --features="ws" --example chat
```
Comment (davidpdrsn) [inline examples/chat.rs:6]: nitpick:

```suggestion
//! cargo run --features=ws --example chat
```

Comment (davidpdrsn) [reply examples/chat.rs:6]: This would also make sense to have for all other example. Doesn't have to be done in this PR though.

**Review by davidpdrsn — CHANGES_REQUESTED** [review]
Looks good!

I had one small nitpick.

**Comment by pepoviola** [comment]
Hi @davidpdrsn, thanks for the review! I already applied the suggestion. 

Is nice to have the `required-features` for examples, I can go over and add for the other examples as well in another pr.

Thanks!

**Review by davidpdrsn — APPROVED** [review]

**Pull request merged**